nmigen_boards/orangecrab_r0_2.py

```diff
@@ -0,0 +1,109 @@
+import os
+import subprocess
+import shutil
+
+from nmigen.build import *
+from nmigen.vendor.lattice_ecp5 import *
+from .resources import *
+
+
+__all__ = ["OrangeCrabR0_2Platform"]
+
+
+class OrangeCrabR0_2Platform(LatticeECP5Platform):
+    device      = "LFE5U-25F"
+    package     = "MG285"
+    speed       = "8"
+    default_clk = "clk"
+    resources   = [
+        Resource("clk", 0, Pins("A9", dir="i"),
+                 Clock(48e6), Attrs(IO_STANDARD="LVCMOS33")),
+        Resource("rst", 0, PinsN("V17"), Attrs(IO_TYPE="LVCMOS33")),
+
+        RGBLEDResource(0, r="K4", g="M3", b="J3", invert=True,
+                      attrs=Attrs(IO_TYPE="LVCMOS33")),
+
+        *ButtonResources(pins={0: "J17" },
+                         attrs=Attrs(IO_TYPE="SSTL135_I")),
+
+        *SPIFlashResources(0,
+                           cs="U17", clk="U16", miso="T18", mosi="U18", wp="R18", hold="N18",
+                           attrs=Attrs(IO_STANDARD="LVCMOS33")
+        ),
+
+        Resource("ddr3", 0,
+            Subsignal("rst",     PinsN("L18", dir="o")),
+            Subsignal("clk",     DiffPairs("J18", "K18", dir="o"), Attrs(IO_TYPE="SSTL135D_I")),
+            Subsignal("clk_en",  Pins("D18", dir="o")),
+            Subsignal("cs",      PinsN("A12", dir="o")),
+            Subsignal("we",      PinsN("B12", dir="o")),
+            Subsignal("ras",     PinsN("C12", dir="o")),
+            Subsignal("cas",     PinsN("D13", dir="o")),
+            Subsignal("a",       Pins("C4 D2 D3 A3 A4 D4 C3 B2 B1 D1 A7 C2 B6 C1 A2 C7", dir="o")),
+            Subsignal("ba",      Pins("P5 N3 M3", dir="o")),
+            Subsignal("dqs",     DiffPairs("G18 H17", "B15 A16", dir="io"),
+                      Attrs(IO_TYPE="SSTL135D_I", TERMINATION="OFF",
+                      DIFFRESISTOR="100")),
+            Subsignal("dq",      Pins("C17 D15 B17 C16 A15 B13 A17 A13 F17 F16 G15 F15 J16 C18"
+                                      "H16 F18", dir="io"), Attrs(TERMINATION="75")),
+            Subsignal("dm",      Pins("G16 D16", dir="o")),
+            Subsignal("odt",     Pins("C13", dir="o")),
+            # top up pins, leave these at their default value
```

Comment:
Is "top up pin" some sort of established name? Because I've never heard of it before OrangeCrab and it doesn't mean anything to me.

Reply:
It was a term I made up on the spot, definitely not an industry standard name

Reply:
Lattice call them "pseudo power pads": http://www.latticesemi.com/en/Support/AnswerDatabase/1/7/3/1736

Reply:
That's much better!

More on topic, I'd probably put these in a separate resource to make it clear that it isn't a part of how DDR3 normally operates. Something like `ddr3_pseudo_power`.